MJxPrep.js

```diff
@@ -270,11 +296,26 @@ if (window.MJxPrep) {
     // default value for startingAt
     var startingAt = startingAt ? startingAt : 0
 
-    // Replace the first instance of 'funcName(<args>)'
-    var funcStart = expr.indexOf(funcName, startingAt)
-    if (funcStart < 0) {
-      return expr
-    }
+    // Find the first instance of 'funcName(<args>)' we care about
+    do {
```

Comment:
Good idea to check for `(` and previous alphabetic characters.

**Opinion:** When reading this, I found the `do...while...` loop hard to reason about, so it took me a bit to understand what was happening. (in particular, `while(true)` especially looks scary, but I found the `break` 😄)

Suggestion: We're already doing this recursively, so how would you feel about replacing the `do` loop with something like:

```js
  function replaceFunctionCalls(expr, funcName, action, startingAt) {
    // default value for startingAt
    var startingAt = startingAt ? startingAt : 0
    var funcStart = expr.indexOf(funcName + '(', startingAt);

    if (funcStart < 0) { return expr; }
    // Make sure the previous character isn't an alpha character
    // (don't match the end of a function name we don't want to match)
    // This will allow us to replace "f(...)" without replacing "diff(...)"
    if( funcStart > 0 && /[a-zA-Z]/.test(expr.substr(funcStart-1, 1)) ) {
      // False positive, look further
      return replaceFunctionCalls(expr, funcName, action, funcStart + 1)
    }

  // etc ...
  }
```

Reply:
Sure, can do that.